igor-monitor-travis/src/main/java/com/netflix/spinnaker/igor/travis/TravisBuildMonitor.java

```diff
@@ -212,9 +212,12 @@ private void processBuild(
         if (!travisService.isLogReady(build)) {
           break;
         }
-        if (build.getNumber()
-            > buildCache.getLastBuild(
-                master, build.getRepository().getSlug(), build.getState().isRunning())) {
+        if (build
+                .getNumber()
+                .compareTo(
```

Comment:
Just a minor concern... since this is looking for making sure there aren't new builds to trigger, the compareTo with this being string vs. int now may not behave exactly the way intended.  I've not checked everyplace on this PR yet, was reading through/glancing through and type conversions where this happens... since it's a string comparison we're going to see very different results...